File: Qytetet Ruby/pkg/QytetetRuby-0.0.1/lib/calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module ModeloQytetet
  class Calle < Casilla
    def initialize(numero,titulo)
      super(numero,TipoCasilla::CALLE,titulo.precio_compra)
      @titulo=titulo
    end
    attr_accessor   :titulo
    def tengo_propietario()
        return  @titulo.tengo_propietario
    end

    def asignar_propietario(jugador)
      @titulo.set_propietario(jugador)  
      return @titulo
    end

    def pagar_alquiler()
      return @titulo.pagar_alquiler      
    end
    
    def propietario_encarcelado()
      return @titulo.propietario.encarcelado      
    end
    def soy_edificable
      return true
    end
    def to_s
      return super + "Titulo propiedad: #{@titulo}"
    end
  end
end

File: Qytetet Java/src/modeloqytetet/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author migue
 */
public class Calle extends Casilla {
    private TituloPropiedad titulo;
    
    Calle(int numeroCasilla,TituloPropiedad titulo){
        super(numeroCasilla);
        super.setCoste(titulo.getPrecioCompra());
        this.titulo=titulo;
    }
    public void asignarPropietario(Jugador jugador){
        this.titulo.setPropietario(jugador);
    }
    
    @Override
    protected TipoCasilla getTipo(){
        return TipoCasilla.CALLE;
    }
    
    @Override
    protected TituloPropiedad getTitulo(){
        return titulo;
    }

    public int pagarAlquiler (){
       return this.titulo.pagarAlquiler();
    }
    
    private void setTitulo(TituloPropiedad titulo){
        this.titulo=titulo;
    }
    @Override
    protected boolean soyEdificable(){
        return true;
    }
    @Override
    public boolean tengoPropietario(){
        return this.titulo.tengoPropietario();
    }
    
    @Override
    public String toString() {
        return super.toString() + "Calle{" + "titulo=" + titulo + '}';
    }
    
}
